### PR TITLE
feat: add 'lost' and 'other' part to the path analysis dashboard

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 1]
```diff
@@ -72,6 +72,24 @@ export interface SQLParameters {
   readonly pairEventAndConditions?: PairEventAndCondition[];
 }
 
+export const buildInEvents = [
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:75]: `builtIn` is better

Comment (zxkane) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:75]: @zhu-xiaowei should we simplfy treat the built-in properties starting with `_`?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts (hunk 1 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:75]: customer may also use '_' as event name first letter.

Comment (zhu-xiaowei) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts:75]: > @zhu-xiaowei should we simplfy treat the built-in properties starting with `_`?

Listing all preset events is more rigorous. We do not restrict users from using events starting with `_`

**Review by zxkane — APPROVED** [review]

File: src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts [hunk 2]
```diff
@@ -3167,208 +3167,647 @@ describe('SQL Builder test', () => {
...
+          app_info.app_id::varchar as app_info_app_id,
+          app_info.id::varchar as app_info_package_id,
+          app_info.install_source::varchar as app_info_install_source,
+          app_info.version::varchar as app_info_version,
+          device.vendor_id::varchar as device_id,
+          device.mobile_brand_name::varchar as device_mobile_brand_name,
+          device.mobile_model_name::varchar as device_mobile_model_name,
+          device.manufacturer::varchar as device_manufacturer,
+          device.screen_width::bigint as device_screen_width,
+          device.screen_height::bigint as device_screen_height,
```
Comment (zhu-xiaowei) [inline src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts:3191]: Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK

File: src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts [hunk 3]
```diff
@@ -3167,208 +3167,647 @@ describe('SQL Builder test', () => {
...
```
Comment (zhu-xiaowei) [inline src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts:3525]: Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK

File: src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts [hunk 4]
```diff
@@ -3394,191 +3833,232 @@ describe('SQL Builder test', () => {
...
```
Comment (zhu-xiaowei) [inline src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts:3886]: Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK

**Review by zhu-xiaowei — APPROVED** [review]

**Pull request merged**